### ISSUE TITLE
DHF 5.0.1: default merge strategy never fires

### ISSUE THREAD

**Issue by crism** [marklogic/marklogic-data-hub#2794]
#### The issue

Short description of the problem: When `"default": true` is used in the `merging` section, the specified strategy never fires.

What behavior are you expecting? The default strategy should work.

#### Tech details

Which Operating System are you using? CentOS 7

Which version of MarkLogic are you using? 9.0-9

Which version of the Data Hub Framework are you using? 5.0.1

Is this a QuickStart UI bug? If so, which browser are you using? No.

#### The devil is in the details...

Here is a step definition:
```javascript
{
  "name": "vendor-mastering",
  "description": "Master vendor stubs",
  "stepDefinitionName": "default-mastering",
  "stepDefinitionType": "MASTERING",
  "options": {
    "collections": [
      "vendors-from-saas", 
      "mastered"
    ],
    "importID": null,
    "matchOptions" : {
      "algorithms" : {
        "algorithm" : []
      },
      "dataFormat": "json",
      "propertyDefs": {
        "properties": [
          {
            "localname": "saasId",
            "name": "saasId"
          }
        ]
      },
      "scoring": {
        "add": [
          {
            "propertyName": "saasId",
            "weight": "50"
          }
        ]
      },
      "thresholds": {
        "threshold": [
          {
            "above": "20",
            "label": "Definitive Match",
            "action": "merge"
          }
        ]
      }
    },
    "mergeOptions" : {
      "algorithms": {
        "custom" : [
          {
            "at" : "/custom-modules/merging/utilities.sjs",
            "function" : "removeNulls",
            "name" : "remove-nulls"
          }
        ]
      },
      "mergeStrategies" : [
        {
          "algorithmRef" : "remove-nulls",
          "name" : "remove-nulls"
        }
      ],
      "merging" : [
        {
          "default" : true,
          "strategy" : "remove-nulls"
        }
      ]
    },
    "outputFormat": "json",
    "sourceDatabase": "agile-mdm-data-hub-STAGING",
    "sourceQuery": "cts.andQuery([cts.collectionQuery('vendors-from-saas'),cts.collectionQuery('mdm-content')])",
    "targetDatabase": "agile-mdm-data-hub-STAGING",
    "targetEntity": "Vendor"
  }
}
```
With that step definition, a logging statement in the specified function never fires. If an explicit merging rule is added:
```javascript
{
  "propertyName" : "address1",
  "strategy" : "remove-nulls"
}
```
then the function issues a log message. It does not matter whether `"default" : true` (no quotation marks) or `"default" : "true"` is used.

**Comment by brucean52** [comment]
JIRA ticket: https://project.marklogic.com/jira/browse/DHFPROD-3096

**Comment by rjrudin** [comment]
@crism I tried to reproduce this on both 5.0.3 and the develop branch and I was not able to. Here's what I tried using the smart-mastering-complete example project.

First, I removed all files from ./data/persons except for the last-name-dob-* files; only needed to test with those.

Then I changed build.gradle to use 5.0.3 DHF. I ran hubInit and mlDeploy to get a fresh install of the app on ML 10.0-2 (though the version of ML shouldn't matter for this ticket).

I then changed merge config in persons.flow.json. I added the following merge strategy (in addition to the retain-single-value strategy already present):

```
{
  "name" : "mergeDob",
  "algorithmRef" : "mergeDob"
}
```

And then I changed the propertyName:DateOfBirth merge block to be this instead:

```
{
  "default" : true,
  "strategy" : "mergeDob"
}
```

I added this line of logging to the start of dob-merge.sjs:

    console.log("mergeDob: " + propertyName);

And then I ran the flow via Gradle. The following was logged by dob-merge.sjs:

```
2019-10-03 12:01:41.882 Info: mergeDob: DateOfBirth
2019-10-03 12:01:41.884 Info: mergeDob: FirstName
2019-10-03 12:01:41.885 Info: mergeDob: LastName
2019-10-03 12:01:41.886 Info: mergeDob: SSN
2019-10-03 12:01:41.888 Info: mergeDob: ZipCode
2019-10-03 12:01:41.889 Info: mergeDob: createdOn
2019-10-03 12:01:41.890 Info: mergeDob: createdBy
```

Note that "Address" isn't there, as it has its own merge block.

I also verified that "algorithmRef":"dobMerge" works as well. So I'm not able to reproduce any problem, at least not on DHF 5.0.3 or on the develop branch.
